src/stream/src/executor/temporal_join.rs

```diff
@@ -271,40 +386,42 @@ impl<S: StateStore, const T: JoinTypePrimitive> TemporalJoinExecutor<S, T> {
                 .set(self.right_table.cache.len() as i64);
             match msg? {
                 InternalMessage::Chunk(chunk) => {
+                    // Compact chunk, otherwise the following keys and chunk rows might fail to zip.
+                    let chunk = chunk.compact();
```

Comment:
Why? `chunk.rows()` should only returns visible rows

Reply:
```rust
let keys = K::build(&self.left_join_keys, chunk.data_chunk())?;
for ((op, left_row), key) in chunk.rows().zip_eq_debug(keys.into_iter()) {
```

We need to align keys with rows, but `keys` can't handle visible opes.

Reply:
I guess we can filter out the empty keys for now (but the assumption is really easy to break, we need further encapsulation for this).

Reply:
Yes, this pattern is quite common(`HashJoin`, `TopN`). We can do a refactor for this pattern later.